lib/requestability_resolver.js

```diff
@@ -0,0 +1,28 @@
+const DeliveryLocationsResolver = require('./delivery-locations-resolver')
+class RequestabilityResolver {
+  static fixItemRequestability (elasticSearchResponse) {
+    elasticSearchResponse.hits.hits
+      .map((hit) => {
+        hit._source.items = hit._source.items.map(function (item) {
+          if (item.electronicLocator) return item
+          if (item.recapCustomerCode && item.recapCustomerCode[0]) {
+            item = DeliveryLocationsResolver.attachRecapDeliveryInfo(item, item.recapCustomerCode[0])
+          } else {
+            item = DeliveryLocationsResolver.attachOnsiteDeliveryInfo(item, item.recapCustomerCode)
```

Comment:
This attaches `.deliveryLocation` to all items including all items served through search results, which could be significant extra bytes to download for some search results (and a violation of our swagger), so ideally we only serve delivery locations through the deliveryLocationsByBarcode endpoint.

Could `DeliveryLocationsResolver` export something like a `.getOnsiteDeliveryInfo(item)` that returns the same properties? (While still exporting these attach functions, which would just be convenience methods to make internal calls to `getOnsiteDeliveryInfo` etc and attach the props to the items?)

Reply:
Also, no second param allowed in `attachOnsiteDeliveryInfo`